docs/reference/cli.cue

```diff
@@ -0,0 +1,230 @@
+package metadata
+
+// These sources produce JSON providing a structured representation of the
+// Vector CLI (commands, flags, etc.)
+
+#Args: [Arg=string]: {
+	description: !=""
+	type:        "string" | "list" | *"string"
+	default?:    string | [...string]
+}
+
+#CommandLineTool: {
+	name:     !=""
+	flags:    #Flags
+	options:  #Options
+	commands: #Commands
+}
+
+#Commands: [Command=string]: {
+	description: !=""
+	flags?:      #Flags
+	options?:    #Options
+	args?:       #Args
+}
+
+#Flags: [Flag=string]: {
+	flag:        "--\(Flag)"
+	description: !=""
+	default:     bool | *false
+
+	if _short != _|_ {
+		short: "-\(_short)"
+	}
+
+	_short: !=""
+}
+
+#Options: [Option=string]: {
+	option:      "--\(Option)"
+	description: !=""
+	default?:    string
+	enum?: [...string]
+
+	if _short != _|_ {
+		short: "-\(_short)"
+	}
+
+	_short: !=""
+}
+
+_default_flags: {
+	"help": {
+		_short:      "h"
+		description: "Prints help information"
+	}
+	"version": {
+		_short:      "V"
+		description: "Prints version information"
+	}
+}
+
+vector: cli: #CommandLineTool & {
```

Comment:
Could this just be `cli:`?

Reply:
Yeah, that works too. I did it this way because in the big metrics PR I have a root `vector` object to which I attach instance-level metrics (i.e. metrics not associated with a component). But we can cross that bridge when we come to it.